### ISSUE TITLE
No way to view doc-strings for types

### ISSUE THREAD

**Issue by brianhuffman** [GaloisInc/cryptol#456]
For value declarations with doc-strings, we can view the help text with `:?` or `:help`:
```
Cryptol> :? max

    max : {a} (Cmp a) => a -> a -> a

Returns the greater of two comparable arguments.
Bitvectors are compared using unsigned arithmetic.
```

But if we create a Cryptol module that has a doc-string on a type synonym declaration, there is apparently no way to view the doc-string:
```
module Byte where
/** A byte is a bit-vector of width 8. **/
type byte = [8]
```
```
Byte> :? byte

[error] at :1:1--1:1
    Expected a value named 'byte' but found a type instead
    Did you mean `(byte)?
Byte> :help byte

[error] at :1:1--1:1
    Expected a value named 'byte' but found a type instead
    Did you mean `(byte)?
Byte> :i byte
Unknown command: :i
```
Values and types are in separate namespaces in Cryptol, so maybe it's not so simple to use the same `:?` command for both; but maybe we could implement a separate `:i` command like ghci has that can print info about a type?

**Comment by yav** [comment]
That's a good idea.  I'll have a stab at fixing this.

**Comment by brianhuffman** [comment]
We should make sure that it works for type constraint synonyms as well. Also it'd be useful to have doc-strings for the primitive type operators (the named ones, at least) but since those don't have declarations in the prelude I'm not sure where the doc-strings would actually go.

**Comment by yav** [comment]
Ok, so implemented support for type synonyms, type constraint synonyms, and newtypes.

I didn't do the built-in types yet, because maybe we should refactor how those are resolved first.

**Comment by brianhuffman** [comment]
Thanks, this looks good. So if there is a type and a value with same name in scope, then it prints out the doc-strings for both; this is probably what we want.

However, I noticed that if I use `:?` with an unknown name, we used to get
```
Cryptol> :? asdf

[error] at :1:1--1:1 Value not in scope: asdf
```
but now we just get no output at all; it just immediately presents another REPL prompt. I think it would be good to keep some kind of "name not in scope" error message.
